### DISCUSSION TITLE
Strapi for e-Commerce

### DISCUSSION THREAD

**Discussion by jplaisir** [strapi/strapi#7040] [Legacy Topics]
I'm in the process of building an e-commerce marketplace with Strapi but we'd like to break things into multiple microservices for each component of the site with no single point of failure. Has anyone from this group have done anything like that with Strapi?  your advice would be greatly appreciated. 

![EdU03](https://user-images.githubusercontent.com/16172782/87690111-077df980-c757-11ea-9f6e-3441a46f50b0.png)

**Comment by daksamit** [comment]
I thought about it in a similar way with vue-storefront integration.
I think a good practice would be to create modules as separated strapi plugins.

**Comment by hihurmuz** [comment]
i used strapi for my fintech blog project. Strapi is fast and easy to use if create project for ecommorce it is probably very good. I think if it is done there should be the following features : 

- Microservices
- GraphQL
- Ready themes ( for Vue, Nuxt, React )
- Easy to add plugin

**Comment by lauriejim** [comment]
To to that you will have to create a Collection Type for Shopping Card, Shipping, etc...
Really easy. A REST API will be generated automatically and you will be able to customise everything to match your custom needs.

**Reply by bamne123** [reply]
How this becomes Microservice setup? Explanation will be great.

**Reply by bamne123** [reply]
As per Microservice each Service has its own DB and it can be deployed Separately. With Collection Types it's just creating a Monolithic project everything in single project, just exposing API's for each collection type.
